src/DevSoc.Eventathon.Attendance/AttendanceRepository.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Dapper;
+using DevSoc.Eventathon.Attendance;
+using DevSoc.Eventathon.Attendance.Models;
+using DevSoc.Eventathon.Calendars.Models;
+
+namespace DevSoc.Eventathon.Data;
+
+public class AttendanceRepository : IAttendanceRepository
+{
+    private readonly IConnectionManager _connectionManager;
+
+    public AttendanceRepository(IConnectionManager connectionManager)
+    {
+        _connectionManager = connectionManager;
+    }
+    
+    public async void RegisterAttendance(AttendanceData attendanceData)
```

Comment:
Don't return `void` from an `async` function - you _must_ return the `Task` unless you are not `await`ing it.
You will end up `await`ing an empty value which will throw an exception.

Reply:
What type of task do I return if it really could just be void though? Like in other functions we have Task<string> for example and I don't think I can just do Task<void>

Reply:
There's a non-generic type `Task` - `Task<void>` isn't valid

Reply:
How do you return that though? No matter what I do it keeps saying 'return type is void' even when I change it to Task and try to return something. If I change it to Task with no return statement it doesn't complain but then makes me question why I need change it to type Task in the first place